Engine/DataFeeds/SubscriptionCollection.cs

```diff
@@ -133,7 +133,13 @@ public bool TryRemove(SubscriptionDataConfig configuration, out Subscription sub
                 return false;
             }
 
-            return dictionary.TryRemove(configuration, out subscription);
+            if (!dictionary.TryRemove(configuration, out subscription))
+            {
+                subscription = null;
+                return false;
+            }
+
+            return dictionary.Any() || _subscriptions.TryRemove(configuration.Symbol, out dictionary);
```

Comment:
I think this should still return true since the subscription was removed.

Great catch on this dictionary!